diff --git a/src/GNUmakefile b/src/GNUmakefile
index 2b2958e..7fa433a 100644
--- a/src/GNUmakefile
+++ b/src/GNUmakefile
@@ -1,9 +1,9 @@
 # Commands to compile Hoard for various targets.
 # Run make (with no arguments) to see the complete target list.
 
-CPPFLAGS = -std=c++14 -O3 -DNDEBUG -ffast-math -fno-builtin-malloc -Wall -Wextra -Wshadow -Wconversion -Wuninitialized
+CPPFLAGS = -std=c++14 -O2 -DNDEBUG -ffast-math -fno-builtin-malloc -Wall -Wextra -Wshadow -Wconversion -Wuninitialized
 #CPPFLAGS = -std=c++14 -g -O0 -ffast-math -fno-builtin-malloc -Wall -Wextra -Wshadow -Wconversion -Wuninitialized
-CXX = clang++
+CXX = g++
 
 # Compute platform (OS and architecture) and build accordingly.
 
