diff --git a/src/GNUmakefile b/src/GNUmakefile
index 1f2babf..50e9387 100644
--- a/src/GNUmakefile
+++ b/src/GNUmakefile
@@ -3,7 +3,7 @@
 
 CPPFLAGS = -std=c++14 -O3 -DNDEBUG -ffast-math -fno-builtin-malloc -Wall -Wextra -Wshadow -Wconversion -Wuninitialized
 #CPPFLAGS = -std=c++14 -g -O0 -ffast-math -fno-builtin-malloc -Wall -Wextra -Wshadow -Wconversion -Wuninitialized
-CXX = clang++
+CXX = g++
 
 # Compute platform (OS and architecture) and build accordingly.
 
